src/api/comms.js

```diff
@@ -72,6 +80,48 @@ export function put(uri, body) {
   });
 }
 
+export function patchAddSecretBody(secretName) {
+  const patchAddBody = [
+    {
+      op: 'add',
+      path: 'serviceaccount/secrets/-',
+      value: {
+        name: secretName
+      }
+    }
+  ];
+
+  return patchAddBody;
+}
+
+export function patchRemoveSecretBody(indexOfSecret) {
```

Comment:
Why's it by index and not secret name? Can you add a comment to justify this? We could instead, for example, accept a secret name here and "internally" get its index through our own code so this is a nicer API for others to use

Reply:
The way to remove items through the KubeApi doesnt allow you to remove with a name, it must be index. We have the function indexOfSecret(sa, secretName) which gets the index of the secret name to remove